### ISSUE TITLE
🐛 Ontop materialize - Column Index out of range, 2 > 1. 

### ISSUE THREAD

**Issue by jatoledo** [ontop/ontop#442]
Hi, I have an error when materializing with Ontop 4.1.0

```bash
./ontop materialize -m tmp.ttl -p configuration.properties
```

```bash
08:37:26.061 |-INFO  in i.u.i.o.a.r.impl.QuestQueryProcessor - Ontop has completed the setup and it is ready for query answering!
Loading class `com.mysql.jdbc.Driver'. This is deprecated. The new driver class is `com.mysql.cj.jdbc.Driver'. The driver is automatically registered via the SPI and manual loading of the driver class is generally unnecessary.
08:37:26.386 |-ERROR in i.u.i.o.m.i.DefaultMaterializedGraphResultSet - Problem materializing the class/property http://www.w3.org/ns/r2rml#example/1
java.lang.RuntimeException: org.eclipse.rdf4j.query.QueryEvaluationException: it.unibz.inf.ontop.exception.OntopConnectionException: it.unibz.inf.ontop.exception.OntopConnectionException: java.sql.SQLException: Column Index out of range, 2 > 1.
        at it.unibz.inf.ontop.cli.OntopMaterialize.runWithSingleFile(OntopMaterialize.java:159)
        at it.unibz.inf.ontop.cli.OntopMaterialize.run(OntopMaterialize.java:109)
        at it.unibz.inf.ontop.cli.Ontop.main(Ontop.java:20)
Caused by: org.eclipse.rdf4j.query.QueryEvaluationException: it.unibz.inf.ontop.exception.OntopConnectionException: it.unibz.inf.ontop.exception.OntopConnectionException: java.sql.SQLException: Column Index out of range, 2 > 1.
        at it.unibz.inf.ontop.rdf4j.materialization.impl.DefaultRDF4JMaterializer$GraphMaterializationIteration.hasNext(DefaultRDF4JMaterializer.java:249)
        at org.eclipse.rdf4j.common.iteration.IterationWrapper.hasNext(IterationWrapper.java:63)
        at it.unibz.inf.ontop.cli.OntopMaterialize.serializeTripleBatch(OntopMaterialize.java:242)
        at it.unibz.inf.ontop.cli.OntopMaterialize.runWithSingleFile(OntopMaterialize.java:152)
        ... 2 more

Caused by: it.unibz.inf.ontop.exception.OntopConnectionException: it.unibz.inf.ontop.exception.OntopConnectionException: java.sql.SQLException: Column Index out of range, 2 > 1.
        at it.unibz.inf.ontop.answering.resultset.impl.AbstractTupleResultSet.buildConnectionException(AbstractTupleResultSet.java:139)
        at it.unibz.inf.ontop.answering.resultset.impl.AbstractTupleResultSet.hasNext(AbstractTupleResultSet.java:101)
        at it.unibz.inf.ontop.materialization.impl.DefaultMaterializedGraphResultSet.hasNext(DefaultMaterializedGraphResultSet.java:132)
        at it.unibz.inf.ontop.rdf4j.materialization.impl.DefaultRDF4JMaterializer$GraphMaterializationIteration.hasNext(DefaultRDF4JMaterializer.java:247)
        ... 5 more
Caused by: it.unibz.inf.ontop.exception.OntopConnectionException: java.sql.SQLException: Column Index out of range, 2 > 1.
        at it.unibz.inf.ontop.answering.resultset.impl.AbstractTupleResultSet.buildConnectionException(AbstractTupleResultSet.java:139)
        at it.unibz.inf.ontop.answering.resultset.impl.DistinctJDBCTupleResultSet.getRawObject(DistinctJDBCTupleResultSet.java:93)
        at it.unibz.inf.ontop.answering.resultset.impl.DistinctJDBCTupleResultSet.computeTupleKey(DistinctJDBCTupleResultSet.java:83)
        at it.unibz.inf.ontop.answering.resultset.impl.DistinctJDBCTupleResultSet.moveCursor(DistinctJDBCTupleResultSet.java:73)
        at it.unibz.inf.ontop.answering.resultset.impl.AbstractTupleResultSet.hasNext(AbstractTupleResultSet.java:97)
        ... 7 more
Caused by: java.sql.SQLException: Column Index out of range, 2 > 1.
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)
        at com.mysql.cj.jdbc.result.ResultSetImpl.checkColumnBounds(ResultSetImpl.java:488)
        at com.mysql.cj.jdbc.result.ResultSetImpl.getObject(ResultSetImpl.java:1104)
        at it.unibz.inf.ontop.answering.resultset.impl.DistinctJDBCTupleResultSet.getRawObject(DistinctJDBCTupleResultSet.java:90)
        ... 10 more
Exception in thread "main" java.lang.AssertionError
        at org.junit.Assert.fail(Assert.java:87)
        at org.junit.Assert.fail(Assert.java:96)
        at it.unibz.inf.ontop.cli.Ontop.main(Ontop.java:32)

```
I'm using the [npd-benchmark](https://github.com/ontop/npd-benchmark/tree/master/data/mysql/original_npd) schema and R2RML Mapping. Attached the mapping(mapping-r2rml.ttl) generated by the engine and my tmp.ttl test
[mapping-r2rml.ttl.txt](https://github.com/ontop/ontop/files/6762952/mapping-r2rml.ttl.txt)
[tmp.ttl.txt](https://github.com/ontop/ontop/files/6762958/tmp.ttl.txt) In both cases the engine generates an error.

**Comment by bcogrel** [comment]
Hi @jatoledo, thanks a lot for reporting the issue!

We will need to dive into it.

Best,
Benjamin

**Comment by kontchakov** [comment]
Hi @jatoledo 

I installed a fresh copy of NPD, but cannot reproduce your bug with the latest development version and your tmp.ttl.txt. Could you please provide a more specific example (including the minimal database instance)?

Thanks
Roman

**Comment by jatoledo** [comment]
Hi @kontchakov
I'm using this SQL [schema ](https://github.com/ontop/npd-benchmark/blob/master/data/mysql/original_npd/npd.mysql) from NPD. Are you using the same schema? It has some instances.

My properties :
```bash
jdbc.url = jdbc:mysql://localhost:3306/npd
jdbc.user = root
jdbc.password = root
jdbc.driver = com.mysql.jdbc.Driver
ontop.distinctResultSet=true
```
If you need more information, let me know.
Thanks!
Jhon T.

**Comment by bcogrel** [comment]
Hi @jatoledo, 

Thanks for sharing the properties. I can see you are using a [deprecated](https://ontop-vkg.org/guide/advanced/configuration property (`ontop.distinctResultSet`). Could you please give a try without it?

I think it would make sense for us to remove the corresponding class for 4.2.0.

Best,
Benjamin

**Comment by kontchakov** [comment]
Hi @jatoledo, @bcogrel 

I have managed to reproduce the bug and can confirm that it goes away when ontop.distinctResultSet=true is removed.

Best
Roman

**Comment by jatoledo** [comment]
Thanks, @bcogrel and @kontchakov
 It works!

Jhon T.

**Comment by kontchakov** [comment]
Bug fix with 8b053d2be552c518965329bb2dcb1de363375447.

**Comment by bcogrel** [comment]
The `DistinctJDBCTupleResultSet` and the corresponding option have been removed (https://github.com/ontop/ontop/commit/84deb6f9c3d6f80c55f45e18c64e478320b1bab2) .
